src/ucp/tag/eager_snd.c

```diff
@@ -84,19 +114,21 @@ static size_t ucp_tag_pack_eager_middle_contig(void *dest, void *arg)
     length         = ucp_ep_config(req->send.ep)->max_am_bcopy - sizeof(*hdr);
     ucs_debug("pack eager_middle paylen %zu", length);
     hdr->super.tag = req->send.tag;
-    memcpy(hdr + 1, req->send.buffer + req->send.state.offset, length);
+    ucp_tag_pack_eager_dt_copy(hdr + 1, req->send.buffer, req->send.state.offset,
```

Comment:
for iov, we need to keep iov_it in req->send.state. otherwise, would have to scan the whole list of iov's for every fragment - O(n^2)